src/note/note_type.rs

```diff
@@ -0,0 +1,113 @@
+use group::GroupEncoding;
+use halo2_proofs::arithmetic::CurveExt;
+use pasta_curves::pallas;
+use std::hash::{Hash, Hasher};
+
+use subtle::{Choice, ConstantTimeEq, CtOption};
+
+use crate::constants::fixed_bases::{VALUE_COMMITMENT_PERSONALIZATION, VALUE_COMMITMENT_V_BYTES};
+use crate::keys::IssuerValidatingKey;
+
+/// Note type identifier.
+#[derive(Clone, Copy, Debug, Eq)]
+pub struct NoteType(pallas::Point);
+
+pub const MAX_ASSET_DESCRIPTION_SIZE: usize = 512;
+
+// the hasher used to derive the assetID
+#[allow(non_snake_case)]
+fn assetID_hasher(msg: Vec<u8>) -> pallas::Point {
```

Comment:
Use a snake-case name, e.g. `asset_id_hasher`, and remove the clippy lint suppression.

Reply:
done